src/websocket/stream.py

```diff
@@ -210,53 +239,47 @@ async def transcribe_chunk_partial(
-            # Therefore we just overwrite it
-            if len(self.agreement.get_confirmed_text()) > 0:
-                text = self.agreement.get_confirmed_text()
-
             result = json.dumps({"partial": text}, indent=2)
 
-            self.overall_transcribed_bytes += recent_cache
-
-            await websocket.send_text(result)
+            if not skip_send and not self.close_stream:
+                await websocket.send_text(result)
 
             end_time = time.time()
             self.logger.debug(
                 "Partial transcription took {:.2f} s".format(end_time - start_time)
             )
 
+            # adjust time between transcriptions
+            self.update_partial_threshold(end_time - start_time)
+
         except Exception:
             self.logger.error(
                 "Error while transcribing audio: {}".format(traceback.format_exc())
             )
             self.close_stream = True
 
-    def adjust_threshold_on_latency(self):
-        """Adjusts the partial threshold based on the latency"""
-        seconds_received = len(self.overall_audio_bytes) / BYTES_PER_SECOND
-        seconds_transcribed = len(self.overall_transcribed_bytes) / BYTES_PER_SECOND
-        # Calculate the latency of the connection
-        seconds_difference = seconds_received - seconds_transcribed
-
-        # we need to exclude the partial threshold from the calculation, because if the threshold is reached, the latency is always bad
-        seconds_difference_exclude_partial = (
-            seconds_difference - self.partial_treshold / BYTES_PER_SECOND
-        )
-
-        if seconds_difference_exclude_partial > FINAL_TRANSCRIPTION_TIMEOUT - 2:
-            self.partial_treshold = min(
-                self.partial_treshold * 1.5, self.final_treshold
-            )
-        elif seconds_difference_exclude_partial < FINAL_TRANSCRIPTION_TIMEOUT / 6:
-            self.partial_treshold = max(
-                self.partial_treshold * 0.75,
-                BYTES_PER_SECOND * PARTIAL_TRANSCRIPTION_TIMEOUT,
-            )
+    def update_partial_threshold(self, last_run_duration: float):
+        # dont adjust any timings with a small window
+        # these adjustments would be overwritten anyway
+        if len(self.sliding_window) < MAX_WINDOW_SIZE_BYTES * 0.75 and last_run_duration < self.partial_transcription_byte_threshold / BYTES_PER_SECOND:
+            self.logger.info(f"Current window too small for adjustment ({len(self.sliding_window)}/{MAX_WINDOW_SIZE_BYTES * 0.75})")
+            return
+        new_threshold = (last_run_duration * BYTES_PER_SECOND) + 0.5
+        self.logger.info(f"Adjusted threshold duration to : {new_threshold / BYTES_PER_SECOND}")
+        self.partial_transcription_byte_threshold = new_threshold
+        self.final_publish_second_threshold = self.partial_transcription_byte_threshold * FINAL_PUBLISH_SECOND_THRESHOLD_FACTOR
 
     def export_transcription_and_wav(self):
```

Comment:
wasn't a change of yours, but the wav exporting can be hella big and blocking or not?

Maybe use async versions if available

Reply:
Yes this could be hella blocking. However I think blocking the transcription is not necessarily bad in this case. If anything fails and the runner is already "gone" this would likely be hard to debug/handle.